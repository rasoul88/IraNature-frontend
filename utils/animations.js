import { keyframes } from "styled-components";

export const moveInLeft = keyframes`
    0% {
        opacity: 0;
        transform: translateX(-10rem);
    }

    80% {
        transform: translateX(1rem);
    }

    100% {
        opacity: 1;
        transform: translateX(0);
    }
`;

export const moveInRight = keyframes`
    0% {
        opacity: 0;
        transform: translateX(10rem);
    }

    80% {
        transform: translateX(-1rem);
    }

    100% {
        opacity: 1;
        transform: translateX(0);
    }
`;

export const moveInBottom = keyframes`
    0% {
        opacity: 0;
        transform: translateY(3rem);
    }

    100% {
        opacity: 1;
        transform: translateY(0);
    }
`;

export const Shake = keyframes`
    0% {
        transform: translateX(-.3rem);
    }

    100% {
        transform: translateX(.3rem);
    }
`;

export const pageInRight = keyframes`
    0% {
        opacity: 0;
        transform: translateX(200%);
    }

    80% {
        transform: translateX(-10rem);
    }

    100% {
        opacity: 1;
        transform: translateX(0);
    }
`;
