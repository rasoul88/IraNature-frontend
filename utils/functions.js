import { toast } from "react-toastify";
import Spin from "../public/assets/icons/Spin.svg";

export const hexTorgba = (color, alpha) => {
  if (alpha === undefined) alpha = 1;
  const r = parseInt(color.substr(1, 2), 16);
  const g = parseInt(color.substr(3, 2), 16);
  const b = parseInt(color.substr(5, 2), 16);
  return `rgba(${r}, ${g}, ${b}, ${alpha})`;
};

export const createGradientbackground = (from, to) => {
  return `linear-gradient(to right bottom, ${from},${to})`;
};

export const createGradientText = (from, to) => {
  return `linear-gradient(to right bottom,${hexTorgba(from, 0.85)}, 
  ${hexTorgba(to, 0.85)})`;
};

export const showToast = (mode, message, onClose) => {
  toast.dismiss();
  toast[mode](message, {
    autoClose: 5000,
    theme: "colored",
    // onOpen: (props) => console.log("i opened"),
    onClose: onClose,
    style: {
      fontFamily: "inherit",
    },
    position: toast.POSITION.BOTTOM_CENTER,
  });
};

export const showLoadingToast = (message) => {
  toast.dismiss();
  toast.info(message, {
    autoClose: false,
    style: {
      fontFamily: "inherit",
      boxShadow: "0 1px 10px 0 rgb(0 0 0 / 30%), 0 2px 15px 0 rgb(0 0 0 / 25%)",
      color: "#333",
    },
    position: toast.POSITION.BOTTOM_CENTER,
    icon: <Spin style={{ transform: "scale(1.5)", fill: "#666" }} />,
  });
};

export const objectDateToString = (objDate) => {
  return `${objDate.year}/${objDate.month < 10 ? "0" : ""}${objDate.month}/${
    objDate.day < 10 ? "0" : ""
  }${objDate.day}`;
};

// export const stringDateToObject = (stringDate) => {
//   const arrDate = stringDate.split("/");
//   return { year: arrDate[0], month: arrDate[1], day: arrDate[2] };
// };
