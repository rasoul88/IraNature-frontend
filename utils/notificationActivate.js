import { post } from "axios";

const urlBase64ToUint8Array = (base64String) => {
  var padding = "=".repeat((4 - (base64String.length % 4)) % 4);
  var base64 = (base64String + padding).replace(/\-/g, "+").replace(/_/g, "/");

  var rawData = window.atob(base64);
  var outputArray = new Uint8Array(rawData.length);

  for (var i = 0; i < rawData.length; ++i) {
    outputArray[i] = rawData.charCodeAt(i);
  }
  return outputArray;
};

const configurePushSub = () => {
  if (!("serviceWorker" in navigator)) {
    return;
  }

  var reg;
  navigator.serviceWorker.ready
    .then(function (swreg) {
      reg = swreg;
      return swreg.pushManager.getSubscription();
    })
    .then(function (sub) {
      if (sub === null) {
        // Create a new subscription
        var vapidPublicKey =
          "BKMOXd1ARsXXz97HxAFb6tU2BmVdhDJkvpscv77UWadgtLidzKZUxHMloFHdOx18kuYxHkYUTHGzbW0I_cwwAyE";
        var convertedVapidPublicKey = urlBase64ToUint8Array(vapidPublicKey);
        return reg.pushManager.subscribe({
          userVisibleOnly: true,
          applicationServerKey: convertedVapidPublicKey,
        });
      } else {
      }
    })
    .then((newSub) => {
      if (newSub) return post("/subscriptions", newSub);
    })
    .then((res) => {
      console.log("new subscription", res);
    })
    .catch(function (err) {
      console.log(err);
    });
};

export const askForNotificationPermission = () => {
  Notification.requestPermission((result) => {
    if (result !== "granted") {
    } else {
      configurePushSub();
    }
  });
};
