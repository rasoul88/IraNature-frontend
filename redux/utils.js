import { objectDateToString } from "../utils/functions";
export const createQuery = (filterItems, dataLimits) => {
  let query = "";
  for (let item in filterItems) {
    if (item === "name" && filterItems[item].length > 0)
      query += `name[regex]=${filterItems[item]}&`;

    if (item === "days") {
      if (filterItems[item][0] > dataLimits[item].min)
        query += `duration[gte]=${filterItems[item][0]}&`;
      if (filterItems[item][1] < dataLimits[item].max)
        query += `duration[lte]=${filterItems[item][1]}&`;
    }

    if (item === "price") {
      if (filterItems[item][0] > dataLimits[item].min)
        query += `price[gte]=${filterItems[item][0]}&`;
      if (filterItems[item][1] < dataLimits[item].max)
        query += `price[lte]=${filterItems[item][1]}&`;
    }

    if (
      item === "dateRange" &&
      filterItems[item].from &&
      filterItems[item].to
    ) {
      query += `startDate[gte]=${objectDateToString(filterItems[item].from)}&`;
      query += `startDate[lte]=${objectDateToString(filterItems[item].to)}&`;
    }

    if (item === "startLocation" && filterItems[item] && filterItems[item][0]) {
      query += `startLocation=${filterItems[item][0].name}&`;
    }

    if (
      item === "difficulty" &&
      (filterItems[item].length === 1 || filterItems[item].length === 2)
    ) {
      filterItems[item].forEach((el) => (query += `difficulty=${el}&`));
    }

    if (
      item === "maxParticipants" &&
      filterItems[item][0] < dataLimits[item].max
    ) {
      query += `maxGroupSize[lte]=${filterItems[item][0]}&`;
    }

    if (item === "page") {
      query += `page=${filterItems[item]}&`;
    }
  }

  query = query.slice(0, query.length - 1);

  return query;
};
