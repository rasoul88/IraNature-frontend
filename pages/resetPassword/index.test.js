import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import * as nextRouter from "next/router";

import { findByTestAttribute } from "../../test/utils";
import { UnconnectedResetPassword } from ".";

const setup = (props) => {
  return shallow(<UnconnectedResetPassword {...props} />);
};

describe("renders resetPassword page", () => {
  test("renders if there is no currentUser", () => {
    const mockUseRouter = jest.fn();
    nextRouter.useRouter = mockUseRouter;
    const wrapper = setup({ currentUser: null });
    const component = findByTestAttribute(wrapper, "component-reset-password");

    expect(component.length).toBe(1);
  });

  test("calls router.push method if there is currentUser", () => {
    const mockPushMethod = jest.fn();
    nextRouter.useRouter = jest.fn().mockReturnValue({ push: mockPushMethod });
    shallow(<UnconnectedResetPassword currentUser={{ name: "rasoul" }} />);

    expect(mockPushMethod.mock.calls[0].length).toBe(1);
    expect(mockPushMethod).toHaveBeenCalledWith("/");
  });

  test("snapshot test", () => {
    const wrapper = setup({ currentUser: null });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("renders error texts", () => {
  test("does not render error texts if there is no input empty error", () => {
    React.useState = jest
      .fn()
      .mockReturnValue([{ passErr: false, confirmPassErr: false }, jest.fn()]);
    const wrapper = setup({ currentUser: null });
    const PassErrorTextcomponent = findByTestAttribute(
      wrapper,
      "pass-error-text"
    );
    const ConfirmPassErrorTextcomponent = findByTestAttribute(
      wrapper,
      "confirm-pass-error-text"
    );
    expect(PassErrorTextcomponent.length).toBe(0);
    expect(ConfirmPassErrorTextcomponent.length).toBe(0);
  });

  test("renders pass error text if password input is empty when we submit the form", () => {
    React.useState = jest
      .fn()
      .mockReturnValue([{ passErr: true, confirmPassErr: false }, jest.fn()]);
    const wrapper = setup({ currentUser: null });
    const PassErrorTextcomponent = findByTestAttribute(
      wrapper,
      "pass-error-text"
    );
    const ConfirmPassErrorTextcomponent = findByTestAttribute(
      wrapper,
      "confirm-pass-error-text"
    );
    expect(PassErrorTextcomponent.length).toBe(1);
    expect(ConfirmPassErrorTextcomponent.length).toBe(0);
  });

  test("renders confirm pass error text if password input is empty when we submit the form", () => {
    React.useState = jest
      .fn()
      .mockReturnValue([{ passErr: false, confirmPassErr: true }, jest.fn()]);
    const wrapper = setup({ currentUser: null });
    const PassErrorTextcomponent = findByTestAttribute(
      wrapper,
      "pass-error-text"
    );
    const ConfirmPassErrorTextcomponent = findByTestAttribute(
      wrapper,
      "confirm-pass-error-text"
    );
    expect(PassErrorTextcomponent.length).toBe(0);
    expect(ConfirmPassErrorTextcomponent.length).toBe(1);
  });

  test("renders not match error text if password and confirm password are not equal", () => {
    React.useState = jest
      .fn()
      .mockReturnValue([
        { passErr: false, confirmPassErr: false, notMatchErr: true },
        jest.fn(),
      ]);
    const wrapper = setup({ currentUser: null });
    const PassErrorTextcomponent = findByTestAttribute(
      wrapper,
      "pass-error-text"
    );
    const ConfirmPassErrorTextcomponent = findByTestAttribute(
      wrapper,
      "confirm-pass-error-text"
    );
    const notMatchErrorTextcomponent = findByTestAttribute(
      wrapper,
      "notMatch-error-text"
    );
    expect(PassErrorTextcomponent.length).toBe(0);
    expect(ConfirmPassErrorTextcomponent.length).toBe(0);
    expect(notMatchErrorTextcomponent.length).toBe(1);
  });

  test("renders both error texts if password and confirmPassword input are empty when we submit the form", () => {
    React.useState = jest
      .fn()
      .mockReturnValue([{ passErr: true, confirmPassErr: true }, jest.fn()]);
    const wrapper = setup({ currentUser: null });
    const PassErrorTextcomponent = findByTestAttribute(
      wrapper,
      "pass-error-text"
    );
    const ConfirmPassErrorTextcomponent = findByTestAttribute(
      wrapper,
      "confirm-pass-error-text"
    );
    expect(PassErrorTextcomponent.length).toBe(1);
    expect(ConfirmPassErrorTextcomponent.length).toBe(1);
  });
});

test("call resetPassword action with correct arguments", () => {
  const token = "sometoken";
  const password = "some-password";
  const mockResetPassword = jest.fn();
  // React.useRef = jest.fn().mockReturnValue({ current: { value: password } });
  jest
    .spyOn(React, "useRef")
    .mockImplementation(() => ({ current: { value: password } }));

  nextRouter.useRouter = jest.fn().mockReturnValue({ query: { token } });

  const wrapper = setup({
    currentUser: null,
    resetPassword: mockResetPassword,
  });

  const formComponent = findByTestAttribute(wrapper, "reset-password-form");

  formComponent.simulate("submit", { preventDefault: jest.fn() });

  expect(mockResetPassword).toHaveBeenCalledWith({
    token,
    password,
    confirmPassword: password,
  });
});
