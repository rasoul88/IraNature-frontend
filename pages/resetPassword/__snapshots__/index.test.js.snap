// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders resetPassword page snapshot test 1`] = `
<indexstyles__PageContainer
  data-test="component-reset-password"
>
  <indexstyles__FormContainer>
    <form
      action=""
      data-test="reset-password-form"
      onSubmit={[Function]}
    >
      <SecodaryHeading
        style={
          Object {
            "fontSize": "3rem",
          }
        }
      >
        بازنشانی کلمه عبور
      </SecodaryHeading>
      <indexstyles__InputContainer>
        <indexstyles__InputField
          placeholder="کلمه عبور"
          type="password"
        />
        <test-file-stub />
      </indexstyles__InputContainer>
      <indexstyles__InputContainer>
        <indexstyles__InputField
          placeholder="تکرار کلمه عبور"
          type="password"
        />
        <test-file-stub />
      </indexstyles__InputContainer>
      <Component
        type="submit"
        value="بازنشانی"
      />
    </form>
  </indexstyles__FormContainer>
</indexstyles__PageContainer>
`;
